scripts/receipts.sh

```diff
@@ -104,4 +108,59 @@ function receipts::generate() {
   syft packages "${image}" --output cyclonedx-json --file "${output}"
 }
 
+# Generates syft receipts for each architecture for given oci archives
+function receipts::generate::multi::arch() {
+  local buildArchive runArchive registryPort registryPid localRegistry imageType archiveName imageReceipt
+
+  buildArchive="${1}"
+  runArchive="${2}"
+
+  # This is hard-coded, but a random port could be obtained if needed
+  registryPort=54321
+
+  # Start a local in-memory registry so we can work with the oci archives
+  PORT=$registryPort crane registry serve --insecure > /dev/null 2>&1 &
+  registryPid=$!
+
+  sleep 2
+
+  localRegistry="127.0.0.1:$registryPort"
+
+  # Push the oci archives to the local registry
+  jam publish-stack \
+    --build-ref "$localRegistry/build" \
+    --build-archive $buildArchive \
+    --run-ref "$localRegistry/run" \
+    --run-archive $runArchive
+  
+  # Ensure we can write to the BUILD_DIR
+  if [ $(stat -c %u build) = "0" ]; then
+    sudo chown -R "$(id -u):$(id -g)" "$BUILD_DIR"
+  fi
+
+  for imageType in build run; do
+    archiveName="${imageType}.oci"
+    util::print::title "Generating package SBOM for ${archiveName}"
+
+    for imageArch in $(crane manifest "$localRegistry/$imageType" | jq -r '.manifests[].platform.architecture'); do
+      util::print::info "Generating CycloneDX package SBOM using syft for $archiveName on platform linux/$imageArch"
+
+      imageReceipt="${BUILD_DIR}/${imageType}-${imageArch}-${receiptFilename}"
+
+      # Generate the architecture-specific SBOM from image in the local registry
+      syft packages "registry:$localRegistry/$imageType" \
+        --output cyclonedx-json \
+        --file "$imageReceipt" \
+        --platform "linux/$imageArch"
+
+      # Copy the amd64 image receipt as the primary one generated by this script
```

Comment:
What's the advantage of having the AMD64 script as the primary one vs. just having multiple arch-specific receipts?

Reply:
I don't think its necessary to have both a `<build or run>-receipt.cyclonedx.json` AND `<build or run>-amd64-receipt.cyclonedx.json` that are the same. I think just `<build or run>-amd64-receipt.cyclonedx.json` is sufficient. might just need to tweak the create-release workflow to work with that naming

Reply:
The goal was to make it work for the existing workflows, and since they don't already account for architecture-specific SBOMs it seemed easiest to just copy the amd64 receipts. I agree that this needs to be looked at, but I was hoping to do it in another PR if that is alright.

Reply:
Yeah I'm onboard with solving the multi-arch receipts later. As long as we don't break the existing pipeline, we can always iterate on adding new features like this.

Reply:
@sophiewigmore I think I'm good with this. If you're happy we can resolve this conversation.